Sources/Subs-Attachments.php

```diff
@@ -1108,33 +1096,32 @@ function getAttachMsgInfo($attachID)
  *
  * @return array.
  */
-function getAttachsByMsg($msgID = 0)
+function getAttachsByMsg($msgID)
 {
-	global $modSettings, $smcFunc, $user_info;
+	global $context, $modSettings, $smcFunc, $user_info;
 	static $attached = array();
 
 	if (!isset($attached[$msgID]))
 	{
 		$request = $smcFunc['db_query']('', '
 			SELECT
-				a.id_attach, a.id_folder, a.id_msg, a.filename, a.file_hash, COALESCE(a.size, 0) AS filesize, a.downloads, a.approved, m.id_topic AS topic, m.id_board AS board,
+				a.id_attach, a.id_folder, a.id_msg, a.filename, a.file_hash, COALESCE(a.size, 0) AS filesize, a.downloads, a.approved, m.id_topic AS topic, m.id_board AS board, m.id_member,
 				a.width, a.height' . (empty($modSettings['attachmentShowImages']) || empty($modSettings['attachmentThumbnails']) ? '' : ',
 				COALESCE(thumb.id_attach, 0) AS id_thumb, thumb.width AS thumb_width, thumb.height AS thumb_height') . '
 			FROM {db_prefix}attachments AS a' . (empty($modSettings['attachmentShowImages']) || empty($modSettings['attachmentThumbnails']) ? '' : '
 				LEFT JOIN {db_prefix}attachments AS thumb ON (thumb.id_attach = a.id_thumb)') . '
 				LEFT JOIN {db_prefix}messages AS m ON (m.id_msg = a.id_msg)
 			WHERE a.attachment_type = {int:attachment_type}
-				' . (!empty($msgID) ? 'AND a.id_msg = {int:message_id}' : '') . '',
+				AND a.id_msg ' . (!empty($context['preview_message']) ? 'IN (0, {int:message_id})' : 'AND a.id_msg = {int:message_id}'),
```

Comment:
e1adb58 removed the useless trailing empty string, but this version still has the `AND a.id_msg ` part twice when `$context['preview_message']` isn't empty.